### PR TITLE
OKTA-544449: Adding support for Splunk Cloud edition

### PR DESCRIPTION
**Pull request by tomaszwrobel-okta**
Resolves: OKTA-544449

## Description:
- **What's changed?** Added edition property to settings object for Splunk Cloud log streams
- **Is this PR related to a Monolith release?** Yes, 2023.02.0

### Resolves:

* [OKTA-544449](https://oktainc.atlassian.net/browse/OKTA-544449)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 1]
```diff
@@ -1803,6 +1803,26 @@ For brevity, the following response doesn't include all available properties.
         "description": "Configuration properties specific to Splunk Cloud",
         "type": "object",
         "properties": {
+          "edition": {
+            "title": "Splunk Edition",
+            "description": "Select a Splunk Edition.",
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:1808]:
```suggestion
            "description": "Select a Splunk Cloud instance edition.",
```

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 2]
```diff
@@ -1803,6 +1803,26 @@ For brevity, the following response doesn't include all available properties.
         "description": "Configuration properties specific to Splunk Cloud",
         "type": "object",
         "properties": {
+          "edition": {
+            "title": "Splunk Edition",
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:1807]:
```suggestion
            "title": "Splunk Cloud instance edition",
```
I'm not sure if this is the official title of this feature? We normally capitalize the official feature name. Can you verify what Splunk uses?

**Review by vanngo-okta — COMMENTED** [review]
Looks good! I just have style nits and questions.
Thanks!

File: packages/@okta/vuepress-site/docs/reference/api/log-streaming/index.md [hunk 3]
```diff
@@ -685,6 +686,7 @@ The Splunk Cloud Settings object specifies the configuration for the `splunk_clo
 
 ```json
 {
+  "edition": "aws",
   "host": "acme.splunkcloud.com"
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/log-streaming/index.md:690]:
```suggestion
  "host": "acme.splunkcloud.com",
```

File: packages/@okta/vuepress-site/docs/reference/api/log-streaming/index.md [hunk 4]
```diff
@@ -694,6 +696,7 @@ The Splunk Cloud Settings object specifies the configuration for the `splunk_clo
 
 | Property      | Description                                                  | DataType                                                       | Nullable | Unique | Readonly | MinLength | MaxLength |
 | ------------- | ------------------------------------------------------------ | -------------------------------------------------------------- | -------- | ------ | -------- | --------- | --------- |
+| edition       | The edition of the Splunk Cloud instance. Can be set to one of the values: `aws`, `gcp` or `aws_govcloud`     | String                                    | FALSE    | FALSE   | FALSE     |      -     |     -      |
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/log-streaming/index.md:699]:
```suggestion
| edition       | The edition of the Splunk Cloud instance. Supported values: `aws`, `gcp`, or `aws_govcloud`     | String                                    | FALSE    | FALSE   | FALSE     |      -     |     -      |
```

**Review by vanngo-okta — CHANGES_REQUESTED** [review]
Minor style nits. We can leave "Splunk Edition" as the title of the schema `edition` property if there's no response from Splunk team. Thanks!

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md (hunk 1 above)
Comment (tomaszwrobel-okta) [reply packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:1808]: This description defines labels in the UI, I'd leave it as is for now

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md (hunk 2 above)
Comment (tomaszwrobel-okta) [reply packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:1807]: We still haven't heard back from Splunk re naming (https://okta.slack.com/archives/CPR1T3GG7/p1670617425367029?thread_ts=1670362962.551249&cid=CPR1T3GG7) and the title is defining form field name, I'd leave it as is for now

**Pull request merged**